README.md

```diff
@@ -31,6 +31,101 @@ Application-level tracing for Rust.
 structured, event-based diagnostic information. `tracing` is maintained by the
 Tokio project, but does _not_ require the `tokio` runtime to be used.
 
+## Usage
+
+### In libraries
+
+Libraries should only rely on the `tracing` crate and use the provided macros
+and types to collect whatever information might be useful to downstream consumers.
+
+```toml
+[dependencies]
+tracing = "0.1"
+```
+
+```rust
+use tracing::{debug, error, info, span, trace, warn, Level};
+
+use std::{error::Error, fmt};
+
+#[tracing::instrument]
+pub fn shave(yak: usize) -> Result<(), Box<dyn Error + 'static>> {
+    debug!(
+        message = "hello! I'm gonna shave a yak.",
+        excitement = "yay!"
+    );
+    if yak == 3 {
+        warn!(target: "yak_events", "could not locate yak!");
+        return Err(ShaveError::new(yak, YakError::new("could not locate yak")).into());
+    } else {
+        trace!(target: "yak_events", "yak shaved successfully");
+    }
+    Ok(())
+}
+
+pub fn shave_all(yaks: usize) -> usize {
+    let span = span!(Level::TRACE, "shaving_yaks", yaks_to_shave = yaks);
+    let _enter = span.enter();
+
+    info!("shaving yaks");
+
+    let mut num_shaved = 0;
+    for yak in 1..=yaks {
+        let res = shave(yak);
+        trace!(target: "yak_events", yak, shaved = res.is_ok());
+
+        if let Err(ref error) = res {
+            error!(
+                message = "failed to shave yak!",
+                yak,
+                error = error.as_ref()
+            );
+        } else {
+            num_shaved += 1;
+        }
+
+        trace!(target: "yak_events", yaks_shaved = num_shaved);
```

Comment:
Similarly, maybe use local var shorthand
```suggestion
        trace!(target: "yak_events", yaks_shaved);
```
and rename the `num_shaved` variable